app/database/MetricsDB.scala

```diff
@@ -56,5 +57,58 @@ class MetricsDB(implicit val db: Database) {
       }
 
   def getDistinctNewspaperBooks: Either[ProductionMetricsError, Seq[Option[String]]] =
-    await(db.run(metricsTable.filter(book => !book.newspaperBook.isEmpty && book.newspaperBook =!= "").map(_.newspaperBook).distinct.result))
+    await(db.run(metricsTable.filter(!_.newspaperBook.isEmpty).map(_.newspaperBook).distinct.result))
+
+  def getGroupedWordCounts(implicit filters: MetricsFilters): Either[ProductionMetricsError, List[GroupedWordCount]] = {
+
+    val lowerBoundsToUpperBounds: Map[Int, Int] = Map(
+      0 -> 349,
+      350 -> 649,
+      650 -> 899
+    )
+
+    awaitWithTransformation(db.run(metricsTable
+      .filter(MetricsFilters.metricFilters)
+      .map( metric =>
+        Case
+          If(metric.wordCount between(0, lowerBoundsToUpperBounds.get(0).get)) Then 0
+          If(metric.wordCount between(350, lowerBoundsToUpperBounds.get(350).get)) Then 350
+          If(metric.wordCount between(650, lowerBoundsToUpperBounds.get(650).get)) Then 650
+          If(metric.wordCount >= 900) Then 900
+      )
+      .groupBy(identity)
+      .map{case (lowerBound, metric) => (lowerBound, metric.size)}
+      .result))(dbResult => {
+
+      dbResult.foldRight(List[GroupedWordCount]())((result: (Option[Int], Int), wordCounts: List[GroupedWordCount]) => {
+        result match {
+          case (None, _) => wordCounts
+          case (Some(lowerBound), count) => {
+            wordCounts ::: List(GroupedWordCount((lowerBound, lowerBoundsToUpperBounds.get(lowerBound)), count))
+          }
+        }
+      }).sortWith(_.countRange._1 < _.countRange._1)
+    })
+  }
+
+  def getArticlesWithWordCounts(withCommissionedLength: Boolean)(implicit filters: MetricsFilters):
+    Either[ProductionMetricsError, List[ArticleWordCountResponse]] = {
+
+    val filterFunction = if (withCommissionedLength)
+      MetricsFilters.withCommissionedWordCountFilters
+    else MetricsFilters.withoutCommissionedWordCountFilters
+
+    awaitWithTransformation(db.run(metricsTable.filter(filterFunction)
+      .take(maxNumberOfArticlesToReturn)
```

Comment:
Part of me wonders whether we should do a `COUNT` somewhere too, which means the results omitted would be ok if it landed on 1000 and we could flag how many results were omitted which might be more useful? Although i don't know how tricky that would be ...

Reply:
If that's something that you think would be useful I can easily just fetch all the results from the database, count them and then not send them all to the client.